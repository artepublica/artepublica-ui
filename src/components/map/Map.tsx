import { useBottomTabBarHeight } from '@react-navigation/bottom-tabs';
import { useHeaderHeight } from '@react-navigation/elements';
import { useWindowDimensions } from 'react-native';
import MapView from 'react-native-maps';

function Map(): JSX.Element {
    const { height, width } = useWindowDimensions();
    const bottomTabBarHeight = useBottomTabBarHeight();
    const headerHeight = useHeaderHeight();

    return <MapView testID="mapView" style={{ height: height - headerHeight - bottomTabBarHeight, width }} />;
}

export default Map;
