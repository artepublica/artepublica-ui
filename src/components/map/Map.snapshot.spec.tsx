import { create } from 'react-test-renderer';

import Map from './Map';

describe('<Map />', () => {
    it('default snapshot', () => {
        const json = create(<Map />).toJSON();

        expect(json).toMatchSnapshot();
    });
});
