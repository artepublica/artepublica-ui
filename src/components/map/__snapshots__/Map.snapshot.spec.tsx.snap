// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> default snapshot 1`] = `
<AIRMap
  initialRegion={null}
  onChange={[Function]}
  onLayout={[Function]}
  onMapReady={[Function]}
  onMarkerPress={[Function]}
  region={null}
  style={
    Object {
      "height": NaN,
      "width": 750,
    }
  }
  testID="mapView"
/>
`;
